### PR TITLE
Update inline docs for pick field selected logic

### PR DESCRIPTION
**Pull request by sc0ttkclark**
Docs for logic fixes in #5014 and #5017

Changelog: Update inline docs for pick field selected logic

### CODE REVIEW

File: classes/fields/pick.php [hunk 1]
```diff
@@ -1298,25 +1298,37 @@ public function build_dfv_field_item_data_recurse_item( $item_id, $item_title, $
...
 			}
+		}
 
-			if ( ! $selected ) {
-				// Cast values in array as string.
-				$args->value = array_map( 'strval', $args->value );
+		// If we do not have a key match, the normal values may still match.
+		if ( ! $selected ) {
+			// Cast values in array as string.
+			$values = array_map( 'strval', $values );
```
Comment (pglewis) [inline classes/fields/pick.php]: I'm considering this just another step as part of properly initializing $values.... any reason not to just do this unconditionally above, after initializing and casting $values?  ~L1308?

**Comment by GaryJones** [comment]
I'd consider most of these comments to be redundant.

The additional comments say _what_ the code is doing, but I can see that from the code.

The comments don't really add anything specific about _why_ the code is doing what it is doing. That where the value of comments are.

If anything could be extracted into named private/protected methods, then that would self-document the _intent_ of the code, and comments would still not be generally needed.

References on comments being a _code smell_:

- https://refactoring.guru/smells/comments
- https://medium.com/@fagnerbrack/code-comment-is-a-smell-4e8d78b0415b
- https://blogs.agilefaqs.com/2013/03/07/inline-comments-in-the-code-is-a-smell-but-document-the-why/
- https://simpleprogrammer.com/why-comments-are-stupid-a-real-example/
- and many more.

**Comment by pglewis** [comment]
A lot of good points, but this is something I asked Scott to do for my own edification.  I plan to boil that down somewhat.  

> I'd consider most of these comments to be redundant.

There are areas where I personally disagree with some of the "comments as code smell" opinions.  A good portion of one's time spent in a legacy codebase is finding _where_ as opposed to _what_.  I think obvious comments at regular intervals can be a very useful thing for how we work because you can scan those obvious comments without digesting the code logic to locate the _where_ more quickly.

File: classes/fields/pick.php (hunk 1 above)
Comment (sc0ttkclark) [reply classes/fields/pick.php]: Yeah this sounds good above.

File: classes/fields/pick.php [hunk 2]
```diff
@@ -1305,6 +1305,9 @@ public function build_dfv_field_item_data_recurse_item( $item_id, $item_title, $
 		if ( ! empty( $args->value ) ) {
 			// The value may be a single non-array value.
 			$values = (array) $args->value;
+
+			// Cast values in array as string.
+			$values = array_map( 'strval', $values );
```
Comment (pglewis) [inline classes/fields/pick.php]: This is still casting conditionally for no good reason that I can see.  It shouldn't functionally matter right now as the only other case is an empty array as the code stands, but code changes.  We insist on a strict string comparison on keys later in the code so we should always, unconditionally cast the keys IMO.

Comment (pglewis) [reply classes/fields/pick.php]: The danger IME is someone comes along later and modifies the enclosing `if` so it's now `if (! empty( $args->value )  && now_only_executes_some_of_the_time )` and we start failing to cast again.

Comment (sc0ttkclark) [reply classes/fields/pick.php]: Fixed

**Comment by JoryHogeveen** [comment]
@pglewis @sc0ttkclark What is the status for this one?

File: classes/fields/pick.php [hunk 3]
```diff
@@ -1304,28 +1304,41 @@ public function build_dfv_field_item_data_recurse_item( $item_id, $item_title, $
 			$edit_link = add_query_arg( array( 'pods_modal' => '1' ), $edit_link );
 		}
 
-		// Determine if this is a selected item
+		// Determine if this is a selected item.
+		// Issue history for setting selected: #4753, #4892, #5014
```
Comment (tr1b0t) [inline classes/fields/pick.php]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: a47f55de534fcce5fe0d9ad2ce17205b</sub>

Comment (sc0ttkclark) [reply classes/fields/pick.php]:
```suggestion
		// Issue history for setting selected: #4753, #4892, #5014.
```

**Pull request merged**